wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1493,22 +1493,23 @@ impl crate::Adapter<super::Api> for super::Adapter {
         let format_aspect = crate::FormatAspects::from(format);
         let limits = self.phd_capabilities.properties.limits;
 
-        let sample_flags = if format_aspect.contains(crate::FormatAspects::DEPTH) {
-            limits
-                .framebuffer_depth_sample_counts
-                .min(limits.sampled_image_depth_sample_counts)
+        let limits_counts = if format_aspect.contains(crate::FormatAspects::DEPTH) {
+            limits.sampled_image_depth_sample_counts
         } else if format_aspect.contains(crate::FormatAspects::STENCIL) {
-            limits
-                .framebuffer_stencil_sample_counts
-                .min(limits.sampled_image_stencil_sample_counts)
+            limits.sampled_image_stencil_sample_counts
         } else {
-            limits
-                .framebuffer_color_sample_counts
-                .min(limits.sampled_image_color_sample_counts)
-                .min(limits.sampled_image_integer_sample_counts)
-                .min(limits.storage_image_sample_counts)
+            match format.describe().sample_type {
+                wgt::TextureSampleType::Float { filterable: _ } => {
+                    limits.sampled_image_color_sample_counts
```

Comment:
The documentation for `framebufferColorSampleCounts` mentions that it only applies to floating-point formats.
```suggestion
                    limits
                        .framebuffer_color_sample_counts
                        .min(limits.sampled_image_color_sample_counts)
```